### ISSUE TITLE
react-native-reanimated 3.3.0 crash on production start

### ISSUE THREAD

**Issue by ButuzGOL** [expo/expo#24828]
### Minimal reproducible example

https://github.com/react-navigation/react-navigation/issues/11494

### Summary

Faced crash on only production start app
more details
https://github.com/react-navigation/react-navigation/issues/11494

solution to update for "react-native-reanimated": "3.5.4",
but it throws Error: [Reanimated] Mismatch between JavaScript part and native part of Reanimated (3.5.4 vs 3.3.0).

### Environment

expo-env-info 1.0.5 environment info:
    System:
      OS: macOS 13.0.1
      Shell: 5.8.1 - /bin/zsh
    Binaries:
      Node: 18.17.1 - /usr/local/bin/node
      Yarn: 3.3.1 - /usr/local/bin/yarn
      npm: 9.6.7 - /usr/local/bin/npm
    Managers:
      CocoaPods: 1.12.0 - /usr/local/bin/pod
    SDKs:
      iOS SDK:
        Platforms: DriverKit 22.2, iOS 16.2, macOS 13.1, tvOS 16.1, watchOS 9.1
    IDEs:
      Android Studio: 2022.1 AI-221.6008.13.2211.9619390
      Xcode: 14.2/14C18 - /usr/bin/xcodebuild
    npmPackages:
      expo: ^49.0.9 => 49.0.9 
      react: 18.2.0 => 18.2.0 
      react-dom: 18.2.0 => 18.2.0 
      react-native: 0.72.5 => 0.72.5 
    npmGlobalPackages:
      eas-cli: 3.13.3
      expo-cli: 6.3.10

**Comment by brentvatne** [comment]
this doesn't seem related to any tools in this repository. is this an issue in react-navigation when using reanimated@3.3.0?

**Comment by xn3cr0nx** [comment]
> this doesn't seem related to any tools in this repository. is this an issue in react-navigation when using reanimated@3.3.0?

@brentvatne the problem is actually the reanimated version 3.5.4 still not supported by expo, that requires version 3.3.0. Reanimated fixed some relevant issues after 3.3.0 tough, and it would be really good to make it work. Can we expect compatibility in SDK50?

**Comment by brentvatne** [comment]
you can use any reanimated version that you like! we just recommend 3.3.0 for sdk 49 because it was the latest stable version at the time of release, and it worked well in our integration testing (this is more of a smoke test than thorough test of every feature, we have to trust library maintainers to take care of that). the limitation of only using the recommended version of a native module applies exclusively to Expo Go, [which is only meant for prototyping / quick experiments](https://docs.expo.dev/get-started/expo-go/). if you want to ignore the version validation for reanimated, [see these docs](https://docs.expo.dev/more/expo-cli/#configuring-dependency-validation).

**Comment by ButuzGOL** [comment]
i have tried 
package.json
```
=> "react-native-reanimated": "^3.5.4",

"expo": {
    "install": {
      "exclude": [
        "react-native-reanimated"
      ]
    }
  },
 ```
 and still error
 [Reanimated] Mismatch between JavaScript part and native part of Reanimated (3.5.4 vs. 3.3.0)

**Comment by focux** [comment]
Took me a while to find out why my app was working fine on dev but crashing on production and indeed it was because of the reanimated version — thanks @ButuzGOL

**Comment by ButuzGOL** [comment]
@focux yes it was quite hard to find such case
@brentvatne may be you can suggest another way because with exclude doesnt work ?

**Comment by upp22** [comment]
Ran into this issue today. This originated from the fact I was trying to use expo-router/drawer, which threw with react-native-reanimated at 3.3.0 when using web. Upgrading to 3.5.4 indeed fixed the issue for web (skipped validation as per steps above) but still gave the mismatch error for mobiles (to be honest it could have been the other way around I can't remember - either way it didn't work for both web and devices). So whilst reanimated fixed the issue in 3.5.4, expo still didn't like it and threw the mismatch error. 

I came across a post where bhyoo99 created a patch: 

https://gist.github.com/bhyoo99/f4954316c498f3ca261197166855e21d

if you follow the instructions online for how to use patch-package and apply the patch, it appears to work. It looks like it solves the issue reanimated 3.5.4 addressed, but keeps the version at 3.3.0, which appears to keep expo happy.

I installed postinstall-postinstall + patch-package. Added the patch file to rootdir/patches. 'npx expo install' which applied the patch. Probably/obviously not a long term fix but posted in case its of any value to anyone.
